frontend/coprs_frontend/coprs/views/backend_ns/backend_general.py

```diff
@@ -287,14 +287,28 @@ def build_task_canceled(task_id):
 @backend_ns.route("/pending-actions/")
 def pending_actions():
     'get the list of actions backand should take care of'
+    busy_namespaces = set()
     data = []
+    # waiting repos are ordered
     for action in actions_logic.ActionsLogic.get_waiting():
+        if (
+            action.object_type == "copr"
+            and action.action_type == ActionTypeEnum("delete")
+        ):
+            busy_namespaces.add(action.copr.full_name)
+        elif action.copr.full_name in busy_namespaces:
```

Comment:
Probably last thing;  can `action.copr` be None in any weird case?  I'm curious whether we shouldn't install the "nonnull" database constraint to simplify our life in such cases (we could move the old actions against `copr_id = 0` e.g.)

Reply:
I mean, we never work with the old actions like that; so `copr_id = 0` would be a clear sign that these are the "old actions". And any _new_ actions to be processed would always have `action.copr` working.

Reply:
Or simply `action.copr && action.copr.full_name in busy_namespaces`.

Reply:
good catch! I didn't think about this one. I think `action.copr && action.copr.full_name in busy_namespaces` is good enough for now